core/src/main/java/hudson/PluginManager.java

```diff
@@ -1234,8 +1236,11 @@ private void trackInitialPluginInstall(@Nonnull final List<Future<UpdateCenter.U
         final UpdateCenter updateCenter = jenkins.getUpdateCenter();
 
         updateCenter.persistInstallStatus();
-        if (jenkins.getInstallState() == InstallState.NEW) {
-            jenkins.setInstallState(InstallState.INITIAL_PLUGINS_INSTALLING);
+        if (!Jenkins.getInstance().getInstallState().isSetupComplete()) {
+            if (jenkins.getInstallState() == InstallState.NEW) {
+                jenkins.setInstallState(InstallState.INITIAL_PLUGINS_INSTALLING);
```

Comment:
yeah, this was done before I changed to use `InstallState.proceed...` I'll make sure this gets updated